objects/69ead2/contained/DUESETUP/script.lua

```diff
@@ -0,0 +1,643 @@
+    for _,tbl in pairs(impendTable) do
+        if tbl.energy[3] ~= 0 then
+            count = count + 1
+        end
+    end
+    return count
+end
+
+function g3Plus(card)
+    local selected = g3CardsSelected()
+    local guid = card.guid
+    if impendTable[guid].energy[3] == 1 then
+        impendTable[guid].energy[3] = 0
+    elseif impendTable[guid].energy[3] == -1 or selected < 2 then
+        impendTable[guid].energy[3] = 1
+    end
+    updateEnergyDisplay(guid)
+    updateG3Buttons(guid)
+    updateSave()
+end
+
+function g3Minus(card)
+    local selected = g3CardsSelected()
+    local guid = card.guid
+    if impendTable[guid].energy[3] == -1 then
+        impendTable[guid].energy[3] = 0
+    elseif impendTable[guid].energy[3] == 1 or selected < 2 then
+        impendTable[guid].energy[3] = -1
+    end
+    updateEnergyDisplay(guid)
+    updateG3Buttons(guid)
+    updateSave()
+end
+
+function createAllCardButtons()
+    createAllG3Buttons()
+    for guid,_ in pairs(impendTable) do
+        local card = getObjectFromGUID(guid)
+        if card then
+            Wait.time(function()
+                createCardButtons(card)
+                updateEnergyDisplay(guid)
+                if g3Selected and impendTable[guid].turnSelected ~= Global.getVar("turn") then
+                    createG3Buttons(card)
+                    updateG3Buttons(guid)
+                end
+            end, 0.01)
+        end
+    end
+    updateSave()
+end
+
+------------------
+-- other functions
+------------------
+
+function getButtonIndex(obj,str) -- needed for cards that already have buttons so can be edited normally
+    if obj.getButtons() then
+        for _,tbl in pairs(obj.getButtons()) do
+            if tbl.click_function == str then
+                return tbl.index
+            end
+        end
+    end
+    return nil
+end
+
+function timePasses()
+    local dances = Global.call("getSpirit", {name = spiritName})
+    for _,tbl in pairs(impendTable) do
+        local var = tbl.energy[1] + tbl.energy[2] + tbl.energy[3]
+        if var >= tbl.maxEnergy and tbl.turnSelected ~= Global.getVar("turn") - 1 then
+            clearCardData(tbl.guid)
+        else
+            tbl.energy[1] = var
+            tbl.energy[2] = 0
+            tbl.energy[3] = 0
+            updateEnergyDisplay(tbl.guid)
+        end
+    end
+    g3Selected = true
+    g3Toggle(dances)
+    updateSave()
+end
+
+local function energyPerTurn(dances)
+    local hits = Physics.cast({
+        origin = dances.positionToWorld(energyPoint),
+        direction = Vector(0, 1, 0),
+        max_distance = 1,
+        type = 1, --ray
+        -- debug = true
+    })
+    for _, hit in pairs(hits) do
+        if hit.hit_object.hasTag("Presence") then
+            return 1
+        end
+    end
+    return 2
+end
+
+function onGainPay(params)
+    local dances = Global.call("getSpirit", {name = spiritName})
+    if params.color == Global.call("getSpiritColor", {name = spiritName}) then
```

Comment:
instead of nesting everything inside of this conditional, lets just do an early return instead if it's not true. and also move this above the lookup for the dances object just in case this needs to early return